speedy-recovery-client/src/components/secured/home/HomePageForParent.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+
+class HomePageForParent extends Component {
+  static defaultProps = {
+    event: {
+      start: [],
+      patient: [],
+      user: []
+    }
+  };
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      appointmentTime: {},
+      user: {},
+      patient: {}
+    };
+  }
+
+  componentWillMount() {
+    setTimeout(() => {
+      this.setState({
+        appointmentTime: this.props.event.start,
+        user: this.props.user,
+        patient: this.props.event.patient
+      });
+    }, 1000);
+  }
+  render() {
+    return (
+      <div>
+        <h3>Hello {this.state.user.name}</h3>
+        {new Date(this.state.appointmentTime).toLocaleString("en-uk") !==
+        "Invalid Date" ? (
+          <h4>
+            Here is the time for your child {this.state.patient}'s next
+            appointment:{" "}
+            <Link to={"/secured/calendar"}>
+              {new Date(this.state.appointmentTime).toLocaleString("en-uk")}
+            </Link>
+          </h4>
+        ) : (
+          <h4>You dont't have any appointment</h4>
```

Comment:
Actually, instead of rendering this whole div, could you show a loader instead, before the data is returned?
https://react.semantic-ui.com/elements/loader/

Reply:
Is there anyway that i can check data is returned or not, to determine whether i should load the data or the load loading screen